### ISSUE TITLE
Bug: v2 Transformer Generates optional nested `id` fields for Input types instead of Required

### ISSUE THREAD

**Issue by danrivett** [aws-amplify/amplify-category-api#1428]
### How did you install the Amplify CLI?

npm -g i @aws-amplify/cli

### Amplify CLI Version

11.1.0

### What operating system are you using?

Mac

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No

### Describe the bug

Given the following schema:

```graphql
type Customer @model {
  id: ID!
  firstName: String!
  lastName: String!
  paymentMethods: [PaymentMethod!]!
}

type PaymentMethod {
  id: ID!
  cardMask: String!
  expiryDate: String!
}
```

When running `amplify codegen` the following Input types are generated:

```graphql
input CustomerInput {
  id: ID
  firstName: String!
  lastName: String!
  paymentMethods: [PaymentMethod!]!
}

input PaymentMethodInput {
  id: ID # <-- Here is the problem, id should be required as it's a nested type
  cardMask: String!
  expiryDate: String!
}
```

As shown above the `id` field on both `CustomerInput` and `PaymentMethodInput` are defined as required on the type definitions but optional on the corresponding input definitions.

It makes sense for `CustomerInput.id` to be optional, since it's a top level type and and if not specified it will get automatically generated by Amplify DataStore (or request resolver), but it doesn't make sense for `PaymentMethodInput.id` to be optional, since it doesn't get auto-generated and so can lead to an invalid `PaymentMethod` being created if the `id` field is not specified on the input.

Note: We've been running the latest `@aws-amplify/cli` versions for many months, but this only started happening when we recently attempted again at upgrading to v2 of the GraphQL Transformer (this is our 3rd attempt as each time we ran into bugs or issues such as this one).

This is currently causing compile errors in our integration code which uses these generated types.

### Expected behavior

`id` fields on nested `input` types should not have their type overridden and marked optional, but should remain required.

The following input types should be generated:

```graphql
input CustomerInput {
  id: ID # <-- This can be made optional like it is today
  firstName: String!
  lastName: String!
  paymentMethods: [PaymentMethod!]!
}

input PaymentMethodInput {
  id: ID! # <-- This should remain required
  cardMask: String!
  expiryDate: String!
}
```

### Reproduction steps

See the GraphQL schema above for reproduction steps.

I've also investigated the codebase, and I believe [this commit](https://github.com/aws-amplify/amplify-category-api/commit/12ff663a94a4896bd9eacef3847be15b7631d8df#diff-4450dff424da5e210f2f7f8e70ce1160b3fba5fe14282240573b3a87ef2efb9aR185-R199) introduced the problem back on August 3rd 2021, but I assume the affected codepath isn't active when v1 Transformer is in use since this hasn't been occurring for us running `amplify codegen` until we upgraded to v2.

I highlighted the logic I think the bug occurs in it is in the `fromField()` function in:

[packages/amplify-graphql-transformer-core/src/wrappers/object-definition-wrapper.ts](https://github.com/aws-amplify/amplify-category-api/blob/main/packages/amplify-graphql-transformer-core/src/wrappers/object-definition-wrapper.ts#L152-L166)

It seems to over-simplistically treat all `id: ID!`s as if they are top-level and so auto-generated as necessary, which the above shows isn't true.

```ts
static fromField = (name: string, field: FieldDefinitionNode, document: DocumentNode): InputFieldWrapper => {
  const autoGeneratableFieldsWithType: Record<string, string[]> = {
    id: ['ID'], // <- Annotated: Problem here 
    createdAt: ['AWSDateTime', 'String'], // <- Annotated: No problem here as this can be argued should only be top level (though it's still not 100% true)
    updatedAt: ['AWSDateTime', 'String'], // <- Annotated: Ditto
  };

  let type: TypeNode;

  if (
    Object.keys(autoGeneratableFieldsWithType).indexOf(name) !== -1 &&
    autoGeneratableFieldsWithType[name].indexOf(unwrapNonNull(field.type).name.value) !== -1
  ) {
    // Annotated: The following comment is in the source code and is incorrect, `id`s are not _always_ optional, only top-level ones

    // ids are always optional. when provided the value is used.
    // when not provided the value is not used.
    type = unwrapNonNull(field.type);
  } // ...
};
```

### Log output

<details>

```
# Put your logs below this line


```

</details>

### Additional information

I've added a potential fix for this in #1429.

### Before submitting, please confirm:

- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

**Comment by danrivett** [comment]
@mohitsriv we're attempting our 3rd attempt to migrate over to transformer v2 as we had a space for non-functional work in between functional work.

This bug was one of 3 issues I mentioned late last year that we previously ran into which caused a number of TypeScript compile errors in our codebase as the v2 transformer changes some GraphQL type definitions.

This ticket's description has the details, and I created a PR with a possible solution. Just looping in you in for your info after our discussion last year.

**Comment by AnilMaktala** [comment]
Hi @danrivett, Thank you for raising the issue and submitting the PR. We will investigate the issue and keep you informed of any updates.

**Comment by danrivett** [comment]
Thanks for the update @AnilMaktala.

#1429 has been merged in which fixes this issue, so I'm looking forward to its release, but I'm currently using it via a local build and using `amplify-dev` and is working well.

**Comment by AnilMaktala** [comment]
Hi @danrivett, The merged PR has been included in version 5.3.0, which has been released.
Please refer to release notes [here](https://github.com/aws-amplify/amplify-category-api/releases/tag/%40aws-amplify%2Famplify-category-api%405.3.0).
